x-pack/plugins/canvas/canvas_plugin_src/functions/common/plot/get_flot_axis_config.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, map } from 'lodash';
+// @ts-ignore untyped Elastic library
+import { getType } from '@kbn/interpreter/common';
+import { Style, Ticks, DatatableColumn, AxisConfig, isAxisConfig } from '../../types';
+
+type Position = 'bottom' | 'top' | 'left' | 'right';
+interface Config {
+  show: boolean;
+  position?: Position;
+  min?: number;
+  max?: number;
+  tickSize?: number;
+  ticks?: Array<[]>;
+  mode?: 'time';
+  font?: Style | {};
+}
+
+interface Options {
+  columns?: DatatableColumn[];
+  ticks?: Ticks;
+  font?: Style | {};
+}
+
+export const getFlotAxisConfig = (
```

Comment:
Great, detailed work here! Small note, we'll likely want to phase out Flot, it saw its last commit more than four years ago, iow there's probably no need for further improvement here.

Reply:
@w33ble pointed that out in a chat, so I opted to avoid typing this any further.

Reply:
👍